library/src/main/java/com/owncloud/android/lib/resources/files/ToggleFavoriteRemoteOperation.java

```diff
@@ -89,12 +101,49 @@ protected RemoteOperationResult run(OwnCloudClient client) {
             }
         } catch (IOException e) {
             result = new RemoteOperationResult(e);
-        }  finally {
+        } finally {
             if (propPatchMethod != null) {
                 propPatchMethod.releaseConnection();  // let the connection available for other methods
             }
         }
 
         return result;
     }
+
+    @Override
+    public RemoteOperationResult run(NextcloudClient client) {
+        RemoteOperationResult<Boolean> result;
+
+        List<Property.Name> removeProperties = new ArrayList<>();
+        Map<Property.Name, String> newProperties = new HashMap<>();
+
+        // disable redirect
+        OkHttpClient disabledRedirectClient = client.getClient()
+                .newBuilder()
+                .followRedirects(false)
+                .authenticator(new NextcloudAuthenticator(client.getCredentials(), "Authorization"))
+                .build();
+
+        if (makeItFavorited) {
+            newProperties.put(NCFavorite.NAME, "1");
+        } else {
+            removeProperties.add(NCFavorite.NAME);
+        }
+
+        String webDavUrl = client.getDavUri().toString();
+        String encodedPath = (client.getUserId() + Uri.encode(filePath)).replace("%2F", "/");
+        String fullFilePath = webDavUrl + "/files/" + encodedPath;
+
+        boolean resultCode = false;
+
+        new DavResource(disabledRedirectClient,
+                HttpUrl.get(fullFilePath))
+                .proppatch(newProperties, removeProperties, (response, hrefRelation) -> {
+                    //resultCode = response.isSuccess();
+                });
```

Comment:
We should likely wrap `DavResource` somehow, I don't like how coupled it's becoming to these operations.